lib/puppet/application/apply.rb

```diff
@@ -241,7 +241,7 @@ def main
         end
 
         # Resolve all deferred values and replace them / mutate the catalog
-        Puppet::Pops::Evaluator::DeferredResolver.resolve_and_replace(node.facts, catalog, apply_environment)
+        Puppet::Pops::Evaluator::DeferredResolver.resolve_and_replace(node.facts, catalog, apply_environment, true)
```

Comment:
Do we need to keep this as a parameter, or should we just move all of the deferred resolution to lazy?

Reply:
Ah yeah good question. I was thinking about creating a puppet setting to retain the old behavior by default. Then in Puppet 8 changing the default value to true.